File: backend/src/main/java/com/saif/portfolio/model/Certificate.java
package com.saif.portfolio.model;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "certificates")
@Data                   // Getters + Setters + toString + equals + hashCode
@NoArgsConstructor      // Default constructor
@AllArgsConstructor     // All args constructor
@Builder                // Builder pattern
public class Certificate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(name = "issued_organisation", nullable = false)
    private String issuedOrganisation;

    @Column(name = "issue_date", nullable = false)
    private LocalDate issueDate;

    @Column(name = "media_url", nullable = false, columnDefinition = "TEXT")
    private String mediaUrl;

    @Column(name = "credential_id", nullable = false)
    private String credentialId;

    @Column(name = "credential_url", nullable = false)
    private String credentialUrl;
}

File: frontend/src/types/certificate.ts


export interface Certificate {
  id?:number;
  title: string;                // required, max 255 chars
  issuedOrganisation: string;   // required, max 255 chars
  issueDate: string;            // LocalDate -> ISO string (e.g., "2025-08-31")
  mediaUrl: string;             // required
  credentialId: string;         // required, max 255 chars
  credentialUrl: string;        // required
}


